### PR TITLE
Bug delete account not working

### PR DESCRIPTION
**Pull request by chinezenwosu**
[Trello task](https://trello.com/c/5rYWOdAb/33-delete-account-not-working)

Backend for https://github.com/Our-Leaders/our-leaders-client/pull/73

### CODE REVIEW

File: routes/UserRoutes.js [hunk 1]
```diff
@@ -51,4 +51,10 @@ module.exports = (router) => {
       AuthMiddleware.hasPermission({property: 'users', action: 'delete'}),
       UsersCtrl.deleteAccount
     );
+
+  router.route('/users/:userId/delete-my-account')
+    .post(
```
Comment (bolorundurowb) [inline routes/UserRoutes.js:56]: hey, is there any reason why we'd prefer a `POST` to a `PUT` or `DELETE`? `POST`s are not idempotent hence my question

Comment (chinezenwosu) [reply routes/UserRoutes.js:56]: I could not send password in the body for DELETE and I didn't want to send it as a param. Maybe it doesn't matter tbh

Comment (Saint-lawal) [reply routes/UserRoutes.js:56]: We definitely shouldn't expose the password. Can we use PUT instead?

Comment (chinezenwosu) [reply routes/UserRoutes.js:56]: I'll change it to PUT

Comment (bolorundurowb) [reply routes/UserRoutes.js:56]: True, I agree with Saint, A `PUT` may just solve both problems

**Pull request merged**